File: src/modules/IsVegan.js
import { getNonVeganList } from './IngredientsListProvider';

/**
 * This functions takes the given ingredient
 * and checks it against the non-vegan list of ingredients
 * @param {string} ingredientToCheck - the ingredient to check
 * @return <code>false</code> if ingredient is in nonVeganList or if all ingredient words are in nonVeganList. Otherwise <code>true</code>
 */
export function isVeganIngredient (ingredientToCheck) {
  // true if empty
  if (ingredientToCheck.length === 0) return true;

  const formattedIngredientToCheck = ingredientToCheck.trim().toLowerCase();

  const nonVeganListIncludesIngredient = getNonVeganList().includes(formattedIngredientToCheck);
  if (nonVeganListIncludesIngredient === true) return false;

  const ingredientWords = formattedIngredientToCheck.split(' ');
  if (ingredientWords.length > 1 && ingredientWords.every(word => !isVeganIngredient(word))) return false;

  return true;
}

/**
 * This functions takes a given list of ingredients
 * and checks them against the non-vegan list of ingredients
 * @param {[string]} ingredientsToCheck - the lit of ingredients to check
 * @return <code>true</code> if ingredients are not in nonVeganList
 */
export function isVeganIngredientList (ingredientsToCheck) {
  return !ingredientsToCheck.some(ingredient => !isVeganIngredient(ingredient));
}

/**
 * This functions takes a given list of ingredients
 * and checks them against the non-vegan list of ingredients
 * @param {[string]} ingredientsToCheck - the lit of ingredients to check
 * @return Array of ingredients that are non-vegan
 */
export function containsNonVeganIngredients (ingredientsToCheck) {
  return ingredientsToCheck.filter(
    ingredient => !isVeganIngredient(ingredient)
  );
}

File: python/src/modules/is_vegan.py
from ingredients_list_provider import get_non_vegan_list

def is_vegan_ingredient(ingredient_to_check):
    '''
    This functions takes the given ingredient.
    and checks it against the non-vegan list of ingredients.
    '''
    if(ingredient_to_check == ''):
        return True
    ingredient_to_check = ingredient_to_check.strip().lower().replace(" ", "")
    non_vegan_list = [i.lower().strip().replace(" ","") for i in get_non_vegan_list()]

    if (ingredient_to_check in non_vegan_list):
        return False
    return True

def contains_non_vegan_ingredients(ingredients_to_check):
    '''
    This functions takes a given list of ingredients
    and checks them against the non-vegan list of ingredients.
    '''
    ingredients_to_check = [i.strip().lower().replace(" ", "") for i in ingredients_to_check]
    non_vegan_list = [i.lower().strip().replace(" ","") for i in get_non_vegan_list()]
    return [i for i in ingredients_to_check if i in non_vegan_list]

def is_vegan_ingredient_list(ingredients_to_check):
    '''
    This functions takes a given list of ingredients.
    and checks them against the non-vegan list of ingredients.
    '''
    return not bool([i.lower().strip().replace(" ","") for i in ingredients_to_check if i in get_non_vegan_list()])
